components/FlexibleHeader/tests/unit/FlexibleHeaderTopSafeAreaTests.m

```diff
@@ -0,0 +1,314 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import "../../src/private/MDCFlexibleHeaderTopSafeArea.h"
+#import "MaterialFlexibleHeader.h"
+
+@interface FlexibleHeaderTopSafeAreaTests : XCTestCase
+@end
+
+@interface FlexibleHeaderTopSafeAreaTestsFakeTopSafeAreaDelegate
+    : NSObject <MDCFlexibleHeaderSafeAreasDelegate>
+@property(nonatomic) BOOL isStatusBarShifted;
+@property(nonatomic) BOOL topSafeAreaInsetDidChangeWasCalled;
+@end
+
+@interface FlexibleHeaderTopSafeAreaTestsFakeViewController : UIViewController
+@property(nonatomic) CGFloat topSafeAreaInset;
+@end
+
+@implementation FlexibleHeaderTopSafeAreaTests {
+  MDCFlexibleHeaderTopSafeArea *_topSafeArea;
+  FlexibleHeaderTopSafeAreaTestsFakeTopSafeAreaDelegate *_delegate;
+}
+
+- (void)setUp {
+  [super setUp];
+
+  _topSafeArea = [[MDCFlexibleHeaderTopSafeArea alloc] init];
+  _delegate = [[FlexibleHeaderTopSafeAreaTestsFakeTopSafeAreaDelegate alloc] init];
+}
+
+- (void)tearDown {
+  _topSafeArea = nil;
+  _delegate = nil;
```

Comment:
Is this a practice we should start embracing? I don't think we normally nil-out properties in `tearDown` since without retain cycles, they will go away when the tests complete.